crates/paimon/src/fileindex/file_indexer_factory.rs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::{options::Options, spec::DataType};
+
+use super::FileIndexer;
+
+pub trait FileIndexerFactory: Send + Sync {
+    fn identifier(&self) -> String;
+    fn create(&self, data_type: DataType, options: Options) -> impl FileIndexer;
+}
+
+// pub struct FileIndexerFactoryRegistry<F: FileIndexerFactory> {
```

Comment:
Any usage? If no, please remove these commented codes.

Reply:
The implementation of FactoryUtil here might not be needed at the moment. I plan to remove it in a later part of this PR.